### PR TITLE
fix: apply normalizeToolName() at all tool lookup points

### LINKED ISSUE #11441: Add .trim() to tool name lookup (fixes "Tool not found" for whitespace in model output)
---

Description

When a model outputs a tool call with leading/trailing whitespace in the tool name (e.g., " exec" instead of "exec"), OpenClaw fails to find the tool and returns "Tool not found".

Steps to Reproduce

Use a model that outputs malformed tool names (e.g., openrouter/moonshotai/kimi-k2.5)
Trigger a tool call
Model outputs: {"name": " exec", "arguments": {...}}
OpenClaw returns: Tool  exec not found

Expected Behavior
Tool names should be trimmed before lookup. The existing normalizeToolName() function already does .trim(), but it's not applied at all lookup points.

Suggested Fix

// Before
.find((t) => t.name === toolName)

// After
.find((t) => t.name === toolName?.trim())

Or apply normalizeToolName() consistently at tool dispatch entry points.

Environment

OpenClaw version: 2026.2.2-3
Model: openrouter/moonshotai/kimi-k2.5
14 failures observed across 5 sessions

Workaround

Switch to a model that doesn't output malformed tool names (e.g., Claude).

---

### PR DESCRIPTION
**Pull request by sypsyp97**
## Summary

Apply `normalizeToolName()` consistently at tool dispatch entry points to handle model-generated tool names with leading/trailing whitespace or unexpected casing. Fixes #11441.

## Problem

When a model outputs a tool call with leading/trailing whitespace in the tool name (e.g., `" exec"` instead of `"exec"`), OpenClaw fails to find the tool and returns "Tool not found". The existing `normalizeToolName()` function already handles `.trim()` + `.toLowerCase()` + alias resolution, but was not applied at all lookup points.

## Changes

- **`src/gateway/tools-invoke-http.ts`**: Replace bare `.trim()` with `normalizeToolName()` for the HTTP tools/invoke endpoint — this also adds alias resolution (`bash` → `exec`) and case normalization that `.trim()` alone misses.
- **`src/auto-reply/reply/get-reply-inline-actions.ts`**: Normalize `dispatch.toolName` before tool lookup in skill command dispatch.
- **`src/agents/tool-policy.normalize-tool-name.test.ts`**: Add test coverage for `normalizeToolName()` edge cases — whitespace trimming, case normalization, alias resolution, and empty input handling.

## Upstream note

The core model→tool dispatch in `@mariozechner/pi-agent-core` (`agent-loop.js` `executeToolCalls`) also uses strict `===` equality without trimming:

```js
const tool = tools?.find((t) => t.name === toolCall.name);
```

Fixing that lookup upstream would fully resolve the issue for all model-generated tool calls. This PR addresses the OpenClaw-controlled dispatch points.

## Testing

- [x] New unit tests for `normalizeToolName()` covering whitespace, casing, aliases, empty input
- [x] AI-assisted (OpenClaw agent + Claude Opus 4.6); changes reviewed and understood
- [x] CI: `pnpm build && pnpm check && pnpm test`

<h2>Greptile Overview</h2>

<h3>Greptile Summary</h3>

This PR makes tool dispatch more tolerant of model-generated tool names by applying `normalizeToolName()` (trim + lowercase + alias resolution) at additional entry points:

- `/tools/invoke` HTTP handler now normalizes `body.tool` before policy checks and lookup.
- Inline skill command tool dispatch now normalizes `dispatch.toolName` before searching the available tool list.
- Adds Vitest coverage for `normalizeToolName()` behavior around whitespace, casing, aliases, and empty input.

Overall this improves resilience to minor formatting differences in tool call names, but care is needed to ensure alias normalization doesn’t break lookups when any tools are registered under an aliased (non-canonical) name.

<h3>Confidence Score: 4/5</h3>

- This PR is likely safe to merge, with one correctness check needed around alias vs canonical tool names in HTTP tool invocation.
- Changes are small and targeted (normalizing tool names at dispatch boundaries plus unit tests). The main remaining risk is behavior change when callers send an alias (e.g. "bash") but the actual registered tool name is the alias rather than the canonical name, which would cause a new 404 until lookup/registration is made consistent.
- src/gateway/tools-invoke-http.ts

<sub>(2/5) Greptile learns from your feedback when you react with thumbs up/down!</sub>

**Context used:**

- Context from `dashboard` - CLAUDE.md ([source](https://app.greptile.com/review/custom-context?memory=fd949e91-5c3a-4ab5-90a1-cbe184fd6ce8))
- Context from `dashboard` - AGENTS.md ([source](https://app.greptile.com/review/custom-context?memory=0d0c8278-ef8e-4d6c-ab21-f5527e322f13))

### CODE REVIEW

File: src/gateway/tools-invoke-http.ts [hunk 1]
```diff
@@ -133,7 +133,7 @@ export async function handleToolsInvokeHttpRequest(
   }
   const body = (bodyUnknown ?? {}) as ToolsInvokeBody;
 
-  const toolName = typeof body.tool === "string" ? body.tool.trim() : "";
+  const toolName = typeof body.tool === "string" ? normalizeToolName(body.tool) : "";
```
[the remark this answers is not in the document]
Comment (sypsyp97) [reply src/gateway/tools-invoke-http.ts:177]: All OpenClaw tools are registered with their canonical names (e.g. `"exec"`, not `"bash"`), so `normalizeToolName("bash")` → `"exec"` correctly resolves to the registered tool. The alias map only exists to handle model/user input variations — tool definitions themselves always use canonical names.

Also added an e2e test in f4fb789 that verifies whitespace + casing normalization works end-to-end for `/tools/invoke`.

**Comment by sypsyp97** [comment]
The failing `macos` check is a **pre-existing SwiftFormat lint issue** in `apps/macos/Sources/` (100/204 Swift files need formatting — sortImports, wrapPropertyBodies, docComments, etc.).

This PR only modifies TypeScript files related to `normalizeToolName()` and does not touch any Swift code. All other 19 checks (Node, Bun, Windows, Android, lint, test, protocol) pass successfully.

[Failed job log](https://github.com/openclaw/openclaw/actions/runs/21786660358/job/62859165856)
